sale_coupon_validity_notification/models/coupon.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2023 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, models
+
+
+class CouponCoupon(models.Model):
+    _inherit = "coupon.coupon"
+
+    def write(self, vals):
+        if vals.get("state", False) == "expired":
```

Comment:
If there's no state in the keys this comparison is going to fail
```suggestion
        if vals.get("state", False) == "expired":
```

Reply:
False > "" ?

Reply:
thanks :)